src/preamble.js

```diff
@@ -2179,6 +2179,14 @@ function integrateWasmJS() {
       if (exports.memory) mergeMemory(exports.memory);
       Module['asm'] = exports;
       Module["usingWasm"] = true;
+#if WASM_BACKEND
+      // wasm backend stack goes down
+      STACKTOP = STACK_BASE + TOTAL_STACK;
+      STACK_MAX = STACK_BASE;
+      // can't call stackRestore() here since this function can be called
+      // synchronously before stackRestore() is declared.
+      Module["asm"]["stackRestore"](STACKTOP);
```

Comment:
or, ignore my last comment, doing it here may be nicer, but how about
```
  // We can now set the stack's initial position in the wasm module.
  exports['stackRestore'](STACKTOP);
```
Using `exports` directly clarifies what's going on, I think - we call directly into the exports that we just received.

Reply:
I think i had trouble with the metaDCE discarding the wam stackRestore function in that case.   Or at least with one of the ways I called it.   Will metaDCE see `exports['stackRestore'](STACKTOP);`?

Reply:
Ah, good point. Yes, it's unsafe to capture the exports and access them like I suggested, metadce won't see it.

lgtm!